### PR TITLE
Switch image in gRPC probe tests to agnhost 

### LINKED ISSUE #115781: Switch image in gRPC probe tests to agnhost
- Tests in `test/e2e/common/node/container_probe.go` are using `etcd` image for gRPC probe tests:
  - should not be restarted with a GRPC liveness probe
  - should be restarted with a GRPC liveness probe

We should use `agnhost` image for these tests as it is much smaller size.

/sig node
/kind cleanup
/priority important-soon
/area tests
/help-wanted
/good-first-issue

KEP: https://github.com/kubernetes/enhancements/issues/2727

### PR DESCRIPTION
**Pull request by shogohida**
#### What type of PR is this?
/kind feature 

#### What this PR does / why we need it:
Changes etcd image to agnhost image 
#### Which issue(s) this PR fixes:

Fixes #115781 

#### Does this PR introduce a user-facing change?

```release-note
No
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/node/container_probe.go [hunk 1]
```diff
@@ -1036,27 +1036,26 @@ func runReadinessFailTest(ctx context.Context, f *framework.Framework, pod *v1.P
...
 		ObjectMeta: metav1.ObjectMeta{Name: "test-grpc-" + string(uuid.NewUUID())},
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{
 				{
 					Name:  containerName,
-					Image: imageutils.GetE2EImage(imageutils.Etcd),
+					Image: imageutils.GetE2EImage(imageutils.Agnhost),
 					Command: []string{
-						"/usr/local/bin/etcd",
+						"/usr/local/bin/agnhost",
```
Comment (SergeyKanzhelev) [inline test/e2e/common/node/container_probe.go:1052]: I don't think this will work. Have you tried it?

Comment (SergeyKanzhelev) [reply test/e2e/common/node/container_probe.go:1052]: in this article you can find out how the arguments would look like for the agnhost: https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/

Comment (shogohida) [reply test/e2e/common/node/container_probe.go]: @SergeyKanzhelev 
Thanks for your review! I fixed the code so please check it!

File: test/e2e/common/node/container_probe.go [hunk 2]
```diff
@@ -630,7 +629,7 @@ done
 		podClient := e2epod.NewPodClient(f)
 		terminationGracePeriod := int64(30)
 		script := `
-_term() { 
+_term() {
```
Comment (shogohida) [inline test/e2e/common/node/container_probe.go:632]: Blank space was deleted automatically due to Visual Studio Code.

File: test/e2e/common/node/container_probe.go [hunk 3]
```diff
@@ -641,7 +640,7 @@ trap _term SIGTERM
 touch /tmp/ready
 touch /tmp/liveness
 
-while true; do 
+while true; do
```
Comment (shogohida) [inline test/e2e/common/node/container_probe.go:643]: Blank space was deleted automatically due to Visual Studio Code.

File: test/e2e/common/node/container_probe.go [hunk 4]
```diff
@@ -534,7 +533,7 @@ var _ = SIGDescribe("Probing container", func() {
 			FailureThreshold:    1,
 		}
 
-		pod := gRPCServerPodSpec(nil, livenessProbe, "etcd")
+		pod := gRPCServerPodSpec(nil, livenessProbe, "agnhost")
```
Comment (SergeyKanzhelev) [inline test/e2e/common/node/container_probe.go:536]: did you run the test? You still hasn;t updated the port in liveness probe definition on the line 527

Comment (shogohida) [reply test/e2e/common/node/container_probe.go:536]: Oh sorry about that. I changed it to 5000

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/ok-to-test
/triage accepted
/priority important-soon

**Pull request merged**